lib/miq_expression.rb

```diff
@@ -912,6 +912,32 @@ def self.miq_adv_search_lists(model, what, extra_options = {})
     end
   end
 
+  # MiqExpression.unavailable_fields_for("ChargebackVM")
+  # will return fields which are not related to ChargebackVM.
+  def self.unavailable_fields_for(model)
+    return [] unless Chargeback.db_is_chargeback?(model)
```

Comment:
Is there any way to make this model oriented, so we are not hardcoding constants into MiqExpression?  That is, can MiqExpression ask the model for the available_fields?  (If you're doing this for consistency, then I can understand, but this feels wrong to hardcode this way).